packages/markdown-docx/src/ToCiceroMarkVisitor.js

```diff
@@ -334,14 +364,40 @@ class ToCiceroMarkVisitor {
                         if (runTimeProperties.attributes['w:fill'] === 'F9F2F4') {
                             shadeCodePresent = true;
                         }
+                    } else if (runTimeProperties.name === 'w:vanish') {
+                        vanishPropertyPresent = true;
+                    } else if (runTimeProperties.name === 'w:rFonts') {
+                        if (runTimeProperties.attributes['w:ascii'] === 'Baskerville Old Face') {
+                            fontFamilyPresent = true;
+                        }
                     }
                 }
                 if (colorCodePresent && shadeCodePresent) {
                     nodeInformation.nodeType = TRANSFORMED_NODES.code;
                 }
+                if (vanishPropertyPresent) {
+                    if (fontFamilyPresent) {
+                        nodeInformation.whenSomeType = false;
+                        nodeInformation.hasSome = true;
+                    } else {
+                        nodeInformation.whenSomeType = true;
+                        nodeInformation.hasSome = false;
+                    }
+                } else {
+                    if (fontFamilyPresent) {
+                        nodeInformation.whenSomeType = false;
+                        nodeInformation.hasSome = false;
+                    } else {
+                        nodeInformation.whenSomeType = true;
+                        nodeInformation.hasSome = true;
+                    }
+                }
```

Comment:
Need to understand this before I can give reviews here.

If `hasSome` is `true`, then `whenSome` is rendered and if `hasSome` is false, `whenNone` is rendered. So there are only two possible conditions which should be encompassed by a **single** boolean variable. Why are there two - `nodeInformation.hasSome` and `nodeInformation.whenSomeType`?

The same logic applies to `toOOXML` because you have used two boolean variables there too, `isTraversingFalseCondition` and `isTraversingElseCondition`.

Reply:
@algomaster99 
if `hasSome` is true, then what nodes are `whenSome` type. there is no clear way to distinguish that. That's why we use font properties for determining whether its `whenSome` or `whenNone` type.

Reply:
> if hasSome is true, then what nodes are whenSome type

```json
{
          "$class": "org.accordproject.ciceromark.Optional",
          "hasSome": false,
          "whenSome": [
            {
              "$class": "org.accordproject.commonmark.Text",
              "text": " except for Force Majeure cases in a "
            },
            {
              "$class": "org.accordproject.ciceromark.Variable",
              "value": "100.0",
              "name": "miles",
              "elementType": "Double"
            },
            {
              "$class": "org.accordproject.commonmark.Text",
              "text": " miles radius,"
            }
          ],
          "whenNone": [
            {
              "$class": "org.accordproject.commonmark.Text",
              "text": "This applies even in case a force majeure."
            }
          ],
          "name": "forceMajeure",
          "elementType": "org.accordproject.latedeliveryandpenalty.Distance",
          "nodes": [
            {
              "$class": "org.accordproject.commonmark.Text",
              "text": "This applies even in case a force majeure."
            }
          ]
        }
```
Taking your testcase as example, I think `Optional` node follows an algorithm.

1. If `hasSome` is true, `whenSome` is rendered.
2. If `hasSome` is false, `whenNone` is rendered.

@dselman could you please confirm this? Or else, please link the file where the model of `Optional` is defined.